### DISCUSSION TITLE
Webview is not loading with location path as "/"

### DISCUSSION THREAD

**Discussion by GollaSaiVenkatesh** [microsoft/vscode-discussions#919] [Extension Development QnA] [answered]
I am using vscode webview with solidjs, It is loading with "/index.html" not "/". How to configure webview to land on "/"?

**Comment by r3m0t** [comment]
I don't think that's possible, but your code shouldn't need to look at the `window.location` at all. Can you share a specific error or example repo?

**Comment by GollaSaiVenkatesh** [comment]
@r3m0t I am trying to use `useLocation` from `@solidjs/router` in my webview for routing purposes.

**Reply by r3m0t** [reply]
You can switch to using MemoryRouter or hash-based routing.

**Reply by burkeholland** [reply]
@GollaSaiVenkatesh did this resolve your issue?

**Reply by GollaSaiVenkatesh** [reply]
@burkeholland i didn't get how using MemoryRouter solves this issue. Can you please explain?

**Reply by r3m0t** [reply, marked as answer]
With MemoryRouter you can separate the two things- the URL of the page `window.location` (this is set by VS Code and can't be changed), and the URL used by the router to decide what part of your app to display.

Eg in React, `<MemoryRouter initialEntries> defaults to ["/"] (a single entry at the root / URL)`, there is probably something similar for SolidJS

**Comment by GollaSaiVenkatesh** [comment]
thanks @r3m0t for the support
